enterprise/utils.py

```diff
@@ -166,3 +171,97 @@ def get_catalog_admin_url_template():
+        course: The name of the course or program that the user's been enrolled in
+        course_url: The URL to the LMS details page for the course or program.
+        course_type: The classification of the learning unit ("course" or "program")
+        organization: The name of the organization sponsoring the enrollment
+        start_date: The date the course first becomes available
+        program_type: An additional classifier for the type of the course - for example,
+            a program could be a "MicroMaster" program
+        email_connection: An existing Django email connection that can be used without
+            creating a new connection for each individual message
+    """
+    if hasattr(user, 'first_name') and hasattr(user, 'username'):
+        # PendingEnterpriseCustomerUsers don't have usernames or real names. We should
+        # template slightly differently to make sure weird stuff doesn't happen.
+        user_name = user.first_name
+        if not user_name:
+            user_name = user.username
+    else:
+        user_name = None
+
+    # Users have an `email` attribute; PendingEnterpriseCustomerUsers have `user_email`.
+    if hasattr(user, 'email'):
+        user_email = user.email
+    elif hasattr(user, 'user_email'):
+        user_email = user.user_email
+    else:
+        raise TypeError(_('`user` must have one of either `email` or `user_email`.'))
+
+    msg_context = {
+        'user_name': user_name,
+        'course_name': course,
+        'program_type': program_type,
+        'course_type': course_type,
+        'organization_name': organization,
+        'start_date': start_date,
+        'course_url': course_url
+    }
+    try:
+        current_site = get_current_site(request)
+        site_template = current_site.enterprise_enrollment_template  # pylint: disable=no-member
+    except ObjectDoesNotExist:
+        # We don't have a site, or it doesn't define a template; that's fine
+        plain_msg = render_to_string(
+            'enterprise/emails/user_notification.txt',
+            msg_context
+        )
+        html_msg = render_to_string(
+            'enterprise/emails/user_notification.html',
+            msg_context
+        )
+        subject_line = 'You\'ve been enrolled in {}!'.format(course)
+    else:
+        if site_template.subject_line:
+            subject_line = site_template.subject_line.format(**msg_context)
+        else:
+            subject_line = 'You\'ve been enrolled in {}!'.format(course)
+        if site_template.html_template and site_template.plaintext_template:
+            plain_msg, html_msg = site_template.render_templates(msg_context)
+        else:
+            plain_msg = render_to_string(
+                'enterprise/emails/user_notification.txt',
+                msg_context
+            )
+            html_msg = render_to_string(
+                'enterprise/emails/user_notification.html',
+                msg_context
+            )
```

Comment:
Can you refactor this so that the

```python
+            plain_msg = render_to_string(
 +                'enterprise/emails/user_notification.txt',
 +                msg_context
 +            )
 +            html_msg = render_to_string(
 +                'enterprise/emails/user_notification.html',
 +                msg_context
 +            )
```
code doesn't appear twice?